docs/design/sql/01-type-system.md

```diff
@@ -0,0 +1,157 @@
+# SQL Type System
```

Comment:
The missing part is `LATE` type description: it's not documented here and it's not documented in the code. It's not clear how it interacts with other types, in which situations it may pop up and what is the purpose of it. Do we need it after all? If we are going with the static schema approach, we should know all the types.

Reply:
The `LATE` data type is our internal representation of the type, which we failed to resolve during optimization. E.g. this is applicable to parameters. Hence, it is not a part of a normal type system. Its behavior will be covered in future design documents, specifically - schema resolution document.